control-plane/agents/src/bin/core/volume/specs.rs

```diff
@@ -1055,7 +1088,20 @@ impl SpecOperationsHelper for VolumeSpec {
             }
             VolumeOperation::CreateSnapshot(_) => Ok(()),
             VolumeOperation::DestroySnapshot(_) => Ok(()),
-            VolumeOperation::Resize(_) => todo!(),
+            VolumeOperation::Resize(req_size) => {
+                if self.size > *req_size {
+                    return Err(SvcError::InvalidArguments {});
```

Comment:
Probably better to return a more descriptive error here?

```suggestion
                if self.size >= *req_size {
                    return Err(SvcError::InvalidArguments {});
```

Reply:
This fits the most among existing error codes. Will have to add a new one otherwise. OR better have `InvalidArguments` take a custom message. But that'll require change in all existing usages of `InvalidArguments`

Reply:
You can add new custom error code for resize

Reply:
Will add one named `VolumeShrinkUnsupported`